src/librustc_mir/dataflow/impls/requires_storage.rs

```diff
@@ -0,0 +1,112 @@
+pub use super::*;
+
+use rustc::mir::*;
+use crate::dataflow::BitDenotation;
+
+/// A conservative approximation of whether each local requires storage at each
+/// point.
+///
+/// This is used to limit which locals are kept in the generator state.
+///
+/// Currently a local is consider to possibly require storage if:
+/// * Any part of it has ever been initialized - borrowck should ensure that
+///   this ends up the same as it's ever been used. If we add a way to
+///   reference uninitialized variables then that would also `gen` this.
+///   This would also be a problem if we ran the generator transform after
+///   deaggregation.
```

Comment:
MIR certainly has a way to reference uninitialized variables and we do want to run the generator transform after  deaggregation.